tests/test_halo_data_transformer.py

```diff
@@ -0,0 +1,473 @@
+        -3: (send_boundaries[NORTHEAST], recv_boundaries[NORTHEAST]),
+    }
+
+    N_edge_boundaries = {
+        0: (send_boundaries[NORTH], recv_boundaries[SOUTH]),
+        -1: (send_boundaries[NORTH], recv_boundaries[WEST]),
+        -2: (send_boundaries[NORTH], recv_boundaries[NORTH]),
+        -3: (send_boundaries[NORTH], recv_boundaries[EAST]),
+    }
+
+    specification_x = HaloUpdateSpec(
+        n_points=n_halos,
+        shape=x_quantity.data.shape,
+        strides=x_quantity.data.strides,
+        itemsize=x_quantity.data.itemsize,
+        origin=x_quantity.metadata.origin,
+        extent=x_quantity.metadata.extent,
+        dims=x_quantity.metadata.dims,
+        numpy_module=x_quantity.np,
+        dtype=x_quantity.metadata.dtype,
+    )
+    specification_y = HaloUpdateSpec(
+        n_points=n_halos,
+        shape=y_quantity.data.shape,
+        strides=y_quantity.data.strides,
+        itemsize=y_quantity.data.itemsize,
+        origin=y_quantity.metadata.origin,
+        extent=y_quantity.metadata.extent,
+        dims=y_quantity.metadata.dims,
+        numpy_module=y_quantity.np,
+        dtype=y_quantity.metadata.dtype,
+    )
+
+    exchange_descriptors_x = [
+        HaloExchangeData(
+            specification_x,
+            N_edge_boundaries[rotation][0],
+            rotation,
+            N_edge_boundaries[rotation][1],
+        ),
+        HaloExchangeData(
+            specification_x,
+            NE_corner_boundaries[rotation][0],
+            rotation,
+            NE_corner_boundaries[rotation][1],
+        ),
+    ]
+    exchange_descriptors_y = [
+        HaloExchangeData(
+            specification_y,
+            N_edge_boundaries[rotation][0],
+            rotation,
+            N_edge_boundaries[rotation][1],
+        ),
+        HaloExchangeData(
+            specification_y,
+            NE_corner_boundaries[rotation][0],
+            rotation,
+            NE_corner_boundaries[rotation][1],
+        ),
+    ]
+
+    data_transformer = HaloDataTransformer.get(
+        x_quantity.np, exchange_descriptors_x, exchange_descriptors_y
+    )
+
+    data_transformer.async_pack([x_quantity, x_quantity], [y_quantity, y_quantity])
+    data_transformer.synchronize()
+    # Simulate data transfer
+    data_transformer.get_unpack_buffer().assign_from(
+        data_transformer.get_pack_buffer().array
+    )
+    data_transformer.async_unpack([x_quantity, x_quantity], [y_quantity, y_quantity])
+    data_transformer.finalize()
+
+    # From the copy of the original quantity we rotate data
+    # according to the rotation & slice and insert them bak
+    # this reproduce the multi-buffer strategy
+    rotated_x, rotated_y = rotate_vector_data(
+        quantity.data[N_edge_boundaries[rotation][0]],
+        quantity.data[N_edge_boundaries[rotation][0]],
+        -rotation,
+        quantity.dims,
+        quantity.metadata.np,
+    )
+    targe_quanitty_x.data[N_edge_boundaries[rotation][1]] = rotated_x
```

Comment:
Typo in `target_quantity_x` (similar for y below).

Reply:
Fixed